buildrunner/utils.py

```diff
@@ -213,46 +213,30 @@ def _acquire_flock_open(
     :param exclusive: config exclusive lock (True) or shared lock (False), defaults to True
     :return: opened file object if successful else None
     """
+
+    @timeout_decorator.timeout(
+        seconds=timeout_seconds, timeout_exception=FailureToAcquireLockException
+    )
+    def get_lock(file_obj, flags):
+        portalocker.lock(
+            file_obj,
+            flags,
+        )
+        return file_obj
+
     # pylint: disable=unspecified-encoding,consider-using-with
     file_obj = open(lock_file, mode)
-    pid = os.getpid()
     lock_file_obj = None
-    retry_sleep_seconds = 0.5
-
-    start_time = current_time = time.time()
-    duration_seconds = current_time - start_time
-
-    while duration_seconds < timeout_seconds:
-        try:
-            # The LOCK_NB means non-blocking
-            # More information here:
-            # https://docs.python.org/3/library/fcntl.html#fcntl.flock
-            if exclusive:
-                # Obtain an exclusive lock, blocks shared (read) locks
-                fcntl.flock(file_obj, fcntl.LOCK_EX | fcntl.LOCK_NB)
-            else:
-                # Allow multiple people to read from the file at the same time
-                # If another instance adds an exclusive lock in the save method below,
-                # this will block until the lock is released
-                fcntl.flock(file_obj, fcntl.LOCK_SH | fcntl.LOCK_NB)
-        except (IOError, OSError):
-            # Limit the amount of logs written while waiting for the file lock
-            num_seconds_between_log_writes = 10
-            if (
-                0
-                <= (duration_seconds % num_seconds_between_log_writes)
-                <= retry_sleep_seconds
-            ):
-                logger.info(
-                    f"PID:{pid} waiting for file lock on {lock_file} after {duration_seconds} seconds"
-                )
-        else:
-            lock_file_obj = file_obj
-            break
-        time.sleep(retry_sleep_seconds)
-        duration_seconds = time.time() - start_time
+    pid = os.getpid()
 
-    if lock_file_obj is None:
+    try:
+        flags = (
+            portalocker.LockFlags.EXCLUSIVE
+            if exclusive
+            else portalocker.LockFlags.SHARED
```

Comment:
IMHO, I think this should live within the `get_lock` function, since this is forcing the user of `get_lock` to know internal details of how it is implemented (namely: with portalocker). Passing in a `is_exclusive=True` flag or something like that makes this a lot more "portable".

Reply:
But that's just my opinion, it totally works as-is

Reply:
I moved the setting of the lock flags to `get_lock`, it makes the code a little more readable.